src/compiler/types.ts

```diff
@@ -3782,6 +3793,10 @@ namespace ts {
         value: number;
     }
 
+    export interface BigIntLiteralType extends LiteralType {
+        value: number;

```

Comment:
how is it a `number`? a program using `bigint` needs to compile even if the underlying engine at compile time does not support bigint.

Reply:
It was `bigint` initially, but it is `number` for now so that it can be compiled on the old LKG. See 74fd6d30806ad307cc12e432943318007838df2a.